src/xenia/base/memory_posix.cc

```diff
@@ -85,8 +85,13 @@ void* AllocFixed(void* base_address, size_t length,
                  AllocationType allocation_type, PageAccess access) {
   // mmap does not support reserve / commit, so ignore allocation_type.
   uint32_t prot = ToPosixProtectFlags(access);
-  void* result = mmap(base_address, length, prot,
-                      MAP_PRIVATE | MAP_FIXED | MAP_ANONYMOUS, -1, 0);
+  int flags = 0;
+  if (base_address != nullptr) {
+    flags = MAP_PRIVATE | MAP_FIXED | MAP_ANONYMOUS;
+  } else {
+    flags = MAP_PRIVATE | MAP_ANONYMOUS;
+  }
+  void* result = mmap(base_address, length, prot, flags, -1, 0);
```

Comment:
You can simplify this whole fragment to: 

```
int flags = MAP_PRIVATE | MAP_ANONYMOUS;
if (base_address != nullptr) {
  flags |= MAP_FIXED;
}

Reply:
Done.